mycroft/util/parse.py

```diff
@@ -164,6 +169,571 @@ def extractnumber_en(text):
     return val
 
 
+'''authors: spenrod and connorpenrod
+this function takes a sentence and extracts date and time information from it
+returns a 2-element array [date, leftover_string]
+the 'date' is a python datetime object
+the 'leftover_string' contains all keywords not related to the date or time'''
+
+
+def extractdatetime_en(str, currentDate=datetime.now()):
```

Comment:
A couple minor things:
* Let's be Pythonic and name the function extract_datetime_en().  And yeah, we need to rename the existing extractnumber() to match -- but don't worry about that in this commit.
* Add a wrapper extract_datetime() just like there is on extractnumber() that turns around and calls the appropriate version based on a lang parameter.